Kudu.Services/ServiceHookHandlers/OneDriveHandler.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Threading.Tasks;
+using System.Web;
+using Kudu.Contracts.Settings;
+using Kudu.Contracts.Tracing;
+using Kudu.Core;
+using Kudu.Core.Deployment;
+using Kudu.Core.SourceControl;
+using Kudu.Services.FetchHelpers;
+using Newtonsoft.Json.Linq;
+
+namespace Kudu.Services.ServiceHookHandlers
+{
+    public class OneDriveHandler : IServiceHookHandler
+    {
+        private readonly OneDriveHelper _oneDriveHelper;
+
+        public OneDriveHandler(ITracer tracer,
+                              IDeploymentSettingsManager settings,
+                              IEnvironment environment)
+        {
+            _oneDriveHelper = new OneDriveHelper(tracer, settings, environment);
+        }
+
+        public DeployAction TryParseDeploymentInfo(HttpRequestBase request, JObject payload, string targetBranch, out DeploymentInfo deploymentInfo)
+        {
+            deploymentInfo = null;
+            if (!string.Equals("OneDrive", payload.Value<string>("deployer"), StringComparison.OrdinalIgnoreCase))
```

Comment:
GitHub has a "x-Github-Event" header
Bitbucket rely on UserAgent from Request object that has be start with "Bitbucket"
Dropbox rely on payload, if payload contains property "NewCursor" or "scmType" equals "DropboxV2"
CodePlex use "deployer" from payload
VSO reply on payload, "publisherId" equals to "tfs"

seems like there is no standard way

personally i thought deployer seems like the best option. 

and i am not sure it is a good idea to replying on the uri, e.g onedrive use to call skypedrive, API endpoint could be changed.